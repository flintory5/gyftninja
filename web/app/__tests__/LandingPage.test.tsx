import React from 'react';
import { render, screen } from '@testing-library/react';
import Home from '../page';

describe('Landing Page', () => {
  it('renders the headline', () => {
    render(<Home />);
    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument();
  });

  it('renders the subheadline', () => {
    render(<Home />);
    expect(screen.getByText(/unbelievable gift ideas/i)).toBeInTheDocument();
  });

  it('renders the CTA button', () => {
    render(<Home />);
    expect(screen.getByRole('button', { name: /find the perfect gift/i })).toBeInTheDocument();
  });
}); 