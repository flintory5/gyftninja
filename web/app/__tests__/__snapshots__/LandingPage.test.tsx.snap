// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Landing Page renders the complete landing page structure 1`] = `
<div>
  <div
    class="page"
  >
    <main
      class="main"
    >
      <div
        style="text-align: center; max-width: 600px;"
      >
        <h1
          style="font-size: 3.5rem; font-weight: bold; margin-bottom: 1rem; background: linear-gradient(135deg, #667eea 0%, #764ba2 100%); -webkit-text-fill-color: transparent; background-clip: text;"
        >
          🎁 Gyft Ninja
        </h1>
        <p
          style="font-size: 1.5rem; margin-bottom: 2rem; color: rgb(102, 102, 102); line-height: 1.6;"
        >
          Discover 
          <strong>
            unbelievable gift ideas
          </strong>
           with AI-powered intelligence that knows exactly what your loved ones want
        </p>
        <div
          style="margin-bottom: 3rem;"
        >
          <p
            style="font-size: 1.1rem; color: rgb(136, 136, 136); margin-bottom: 2rem; line-height: 1.7;"
          >
            Stop guessing. Start giving gifts that create unforgettable moments. Our intelligent platform analyzes personalities, interests, and relationships to suggest the perfect gift every time.
          </p>
        </div>
        <div
          class="ctas"
        >
          <button
            class="primary"
            style="background: linear-gradient(135deg, #667eea 0%, #764ba2 100%); font-size: 1.1rem; font-weight: 600; padding: 0px 32px; height: 56px;"
          >
            🎯 Find the Perfect Gift
          </button>
        </div>
      </div>
      <div
        style="display: grid; grid-template-columns: repeat(auto-fit, minmax(250px, 1fr)); gap: 2rem; margin-top: 4rem; max-width: 800px;"
      >
        <div
          style="text-align: center; padding: 2rem; border-radius: 12px; background: var(--gray-alpha-100); border: 1px solid var(--gray-alpha-200);"
        >
          <div
            style="font-size: 2.5rem; margin-bottom: 1rem;"
          >
            🧠
          </div>
          <h3
            style="margin-bottom: 0.5rem; font-size: 1.2rem;"
          >
            AI-Powered Intelligence
          </h3>
          <p
            style="color: rgb(102, 102, 102); font-size: 0.9rem;"
          >
            Advanced algorithms analyze personality traits and preferences
          </p>
        </div>
        <div
          style="text-align: center; padding: 2rem; border-radius: 12px; background: var(--gray-alpha-100); border: 1px solid var(--gray-alpha-200);"
        >
          <div
            style="font-size: 2.5rem; margin-bottom: 1rem;"
          >
            💝
          </div>
          <h3
            style="margin-bottom: 0.5rem; font-size: 1.2rem;"
          >
            Personalized Suggestions
          </h3>
          <p
            style="color: rgb(102, 102, 102); font-size: 0.9rem;"
          >
            Tailored recommendations based on relationships and occasions
          </p>
        </div>
        <div
          style="text-align: center; padding: 2rem; border-radius: 12px; background: var(--gray-alpha-100); border: 1px solid var(--gray-alpha-200);"
        >
          <div
            style="font-size: 2.5rem; margin-bottom: 1rem;"
          >
            ✨
          </div>
          <h3
            style="margin-bottom: 0.5rem; font-size: 1.2rem;"
          >
            Unforgettable Moments
          </h3>
          <p
            style="color: rgb(102, 102, 102); font-size: 0.9rem;"
          >
            Create lasting memories with gifts that truly matter
          </p>
        </div>
      </div>
    </main>
    <footer
      class="footer"
    >
      <p
        style="color: rgb(136, 136, 136); font-size: 0.9rem; text-align: center;"
      >
        © 2024 Gyft Ninja - Making gift-giving an art form
      </p>
    </footer>
  </div>
</div>
`;
